test/tests/functional/pbs_offline_vnodes.py

```diff
@@ -86,39 +87,80 @@ def test_single_mom_hook_failure_affects_vnode(self):
+                           id=self.moms.values()[0].shortname, max_attempts=3,
+                           interval=2)
         j1.set_sleep_time(1000)
         jid = self.server.submit(j1)
-        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
         self.server.expect(JOB, {ATTR_state: 'Q'}, id=jid)
 
         self.server.expect(
             NODE, {ATTR_NODE_state: 'offline'},
-            id=self.moms.values()[0].shortname)
+            id=self.moms.values()[0].shortname, max_attempts=3, interval=2)
+
+        vname = "vnode[0]"
         if self.is_cray is True:
             vlist = []
+            first_vnode = 1
             vnl = self.server.filter(
                 VNODE, {'resources_available.vntype': 'cray_compute'})
             vlist = vnl["resources_available.vntype=cray_compute"]
             # Loop through each compute vnode in the list and check if state =
             # offline
             for v1 in vlist:
+                if first_vnode == 1:
+                    first_vnode += 1
+                    vname = v1
                 # Check that the vnode is in offline state
                 self.server.expect(
                     VNODE, {'state': 'offline'}, id=v1, max_attempts=3,
                     interval=2)
 
         else:
             self.server.expect(
-                NODE, {ATTR_NODE_state: 'offline'}, id='vnode[0]')
+                NODE, {ATTR_NODE_state: 'offline'}, id='vnode[0]',
+                max_attempts=3, interval=2)
             self.server.expect(
-                NODE, {ATTR_NODE_state: 'offline'}, id='vnode[1]')
+                NODE, {ATTR_NODE_state: 'offline'}, id='vnode[1]',
+                max_attempts=3, interval=2)
+
+        mom_host = self.moms.values()[0].shortname
+        pbs_exec = self.server.pbs_conf['PBS_EXEC']
+        pbsnodes_cmd = os.path.join(pbs_exec, 'bin', 'pbsnodes')
+        pbsnodes_reset = pbsnodes_cmd + ' -r ' + mom_host
+
+        # Set mom sync hook timeout to be a low value because if mom fails to
+        # get the hook after disabling it then next sync will happen after
+        # 2 minutes by default.
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'sync_mom_hookfiles_timeout': '5'})
+        self.server.manager(MGR_CMD_SET, HOOK, {'enabled': 'False'}, id="h1")
+        # Make sure that hook has been sent to mom
+        self.server.log_match("successfully sent hook file")
+        self.du.run_cmd(self.server.hostname, cmd=pbsnodes_reset, sudo=True)
+        self.server.delete(jid, wait=True)
+
+        j2 = Job(TEST_USER)
+        j2.set_attributes({ATTR_l + '.select': '1:vnode=' + vname})
+        jid2 = self.server.submit(j2)
+        self.server.expect(NODE, {ATTR_NODE_state: 'free'},
+                           id=self.moms.values()[0].shortname, max_attempts=3,
+                           interval=2)
+        self.server.manager(MGR_CMD_SET, SCHED, {'scheduling': 'True'})

```

Comment:
Since you don't make scheduling false anymore, you probably don't need to set it to true.  Submitting the job should cause a scheduling cycle to run.

Reply:
Yes this step is not needed.